Marlin/planner.cpp

```diff
@@ -709,10 +675,6 @@ void Planner::_buffer_line(const float &a, const float &b, const float &c, const
     }
   #endif
 
-  #if ENABLED(LIN_ADVANCE)
-    const float mm_D_float = SQRT(sq(a - position_float[X_AXIS]) + sq(b - position_float[Y_AXIS]));
-  #endif
-
   const long da = target[X_AXIS] - position[X_AXIS],
```

Comment:
@Sebastianv650 Perhaps you can follow this logic in your previous working version of the code and do the same thing. Because as far as I could see, `position_float` is redundant. A lot of effort goes into setting `position_float` but it's only used in this very limited way.

Here, `da`, `db` are integer equivalents to `a - position_float[X_AXIS]` and `b - position_float[Y_AXIS]`. They only need to be multiplied by `steps_to_mm[]` to get `float` values accurate within a microstep. So, `position_float` is not needed, and the `mm_D_float` value above can be figured from `da` and `db`, which they are below.